Orange/widgets/data/owpreprocess.py

```diff
@@ -1258,6 +1386,13 @@ def setModelData(self, widget, index):
         assert isinstance(params, dict)
         self._model.setData(index, params, ParametersRole)
 
+    def set_max_num_attributes(self, num_data_attributes=None):
+        self.max_num_attributes = num_data_attributes
+        for widget in self.view.widgets():
+            row = self.view.indexOf(widget)
+            index = self.model().index(row, 0)
+            self.setWidgetData(widget, index)
+
```

Comment:
This should not be here (in this class).

In fact the individual preprocessors (edited by this widget) should not (in fact can't) be dependent on the input data.